### PR TITLE
Refactoring/minimesos docker repo subdir

### PR DESCRIPTION
**Pull request by alex-glv**
Moves containersolutions/minimesos-docker images to docker subdir.
Removes static linking of docker binary and mounts hosts docker binary to avoid version mismatch

### CODE REVIEW

**Comment by sadovnikov** [comment]
Hi @alex-glv , why do you want to merge `containersolutions/minimesos-docker` repo into `minimesos`? They have very different release cycles.

File: docker/.gitignore [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# PDT-specific
+.buildpath
+
+# sbteclipse plugin
+.target
+
+# TeXlipse plugin
+.texlipse
+
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (sadovnikov) [inline docker/.gitignore]: is this file needed in the folder? There is already one on top level

File: docker/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+cd ..
+
+cd 0.27.0-0.2.190.ubuntu1404
+
+cd agent
+docker build -t containersol/mesos-agent:0.27.0-0.2.190.ubuntu1404 .
+cd ..
+
+cd master
+docker build -t containersol/minimesos-master:0.27.0-0.2.190.ubuntu1404 .
```
Comment (sadovnikov) [inline docker/build.sh]: This file is very different from https://github.com/ContainerSolutions/minimesos-docker/blob/master/build.sh

**Comment by alex-glv** [comment]
@sadovnikov  minimesos can't really function outside of docker context, I think it's good to keep minimesos requirements under the same project.

**Comment by frankscholten** [comment]
I agree with @alex-glv that it's easier to manage the project when we have a single repo. We do have to define a tagging convention to distinguish Mesos versions and minimesos versions.

How about `$minimesos-version-$mesos-version`?

In that case the tags will be

* `containersol/mesos-agent:0.8.0-0.26` for a _release_ of minimesos
* `containersol/mesos-agent:30cb47-0.26` for a _snapshot_ of minimesos based on a commit
* `containersol/mesos-agent:latest` will point to a _default release_. For instance `containersol/mesos-agent:0.8.0-0.25` because a lot of frameworks use Mesos 0.25.

With this setup we trade off precision in being able to see what has changed in each Docker image with clarity of versioning for end-users.

I think this is a good trade-off because in general we only change Docker images if there is some bug like a missing package. They are not updated that often.

When we adopt this we have to change the build so that MesosAgent and MesosMaster point to images tagged with a specific commit. We can do this by having Gradle create a MANIFEST file which contains the current commit. See https://github.com/ContainerSolutions/minimesos/issues/99

File: docker/build.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+cd base
+docker build -t containersol/mesos-base:latest .
+cd ..
+
+buildVersion "0.25.0-0.2.70.ubuntu1404"
+buildVersion "0.26.0-0.2.145.ubuntu1404"
+buildVersion "0.27.0-0.2.190.ubuntu1404"
+buildVersion "0.27.1-2.0.226.ubuntu1404"
```
Comment (sadovnikov) [inline docker/build.sh]: Thanks! However now `0.27.1-2.0.226.ubuntu1404` folder is missing

**Comment by sadovnikov** [comment]
I agree `minimesos` depends on these images, but these images can exist without any knowledge of `minimesos`. Certainly, their versioning is independent from versioning of `minimesos`. I think merging two repositories will make things just more complicated and auto-building of images in Docker Hub impossible.

Leaving decision to press the merge button up to you ;)

**Comment by alex-glv** [comment]
@sadovnikov can you elaborate what actually will get more complicated?
In dockerhub you specify the source repository and the location of the dockerfiles.
So, we'd have containersol/minimesos/docker

For all I know, minimesos build pipeline doesn't have to care about Docker images, same with dockerhub - it will just look into docker folder. 

I think having folder with the images will give a better overall idea of the project, what it depends on and what's contained in the images.

for now we don't even have to change the versioning scheme.

**Comment by sadovnikov** [comment]
Putting two parts, which have different versioning cycles, together is confusing. 
- we'll be tagging minimesos code and this will tag Dockerfiles, which have nothing to do with minimesos;
- what are we going to do if we find a mistake, let's say in mesos client wrapped script, and have to update Dockerfiles and release them? How are we going to tag minimesos repository? 

Example, ES Framework depends on Elastic Search itself. However their versions are different and they are kept in different repositories. There is not confusion about this. Yes, it's unfair comparison because it's impossible to put these two project together. However I don't see other difference

**Comment by alex-glv** [comment]
I don't know answers to the questions you posed, and you said the ES framework and ES comparison is unfair, so that's irrelevant.

The problem I dealt with was https://github.com/ContainerSolutions/minimesos-docker/issues/7
I am not sure why we would bake the fixed version docker binary into the images, so changing that required adding changes to the MesosAgent to include the binary as a volume. So, now to deploy one, I'd depend on another project change to come out.

**Comment by sadovnikov** [comment]
Is including binary as a volume going to work when OS of the host and container are different? 

About merging the two repositories, seems @frankscholten agrees with you: 2 -1. So you can merge ;)

**Comment by frankscholten** [comment]
@remmelt Thoughts?

**Comment by alex-glv** [comment]
I think my pull request mixes two issues together - docker binary linking and moving minimesos-docker repo, I'll close it
